city_scrapers/spiders/il_pollution_control.py

```diff
@@ -0,0 +1,203 @@
+            ):
+                continue  # Not interested in this event type
+
+            title = item["CalendarTypeDesc"].replace("CANCELLED", "").strip()
+            meeting = Meeting(
+                title=title,
+                description="",  # Too inconsistent to parse accurately
+                classification=self._parse_classification(title),
+                start=self._parse_start(item),
+                end=None,
+                all_day=item["IsFullDay"],
+                time_notes="",
+                location=self._parse_location(item),
+                links=list(),
+                source=self._parse_source(item),
+            )
+
+            meeting["links"] = self._parse_links(meeting)
+            meeting["status"] = self._get_status(
+                meeting, text=" ".join([item['CalendarTypeDesc'], item['Description']]).lower()
+            )
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    def _parse_classification(self, title):
+        """Parse or generate classification from allowed options."""
+        if "Board" in title:
+            return BOARD
+        else:
+            return NOT_CLASSIFIED
+
+    def _parse_start(self, item):
+        return datetime.strptime(item["StartDateTime"], '%m/%d/%Y %I:%M:%S %p')
+
+    def _parse_location(self, item):
+        """Parse or generate location."""
+        text = " ".join([item['Description'], item['Location']]).lower()
+        if "thompson" in text:
+            return {
+                "address":
+                    "James R. Thompson Center - 100 W. Randolph St. Suite 11-500, Chicago, IL 60601",  # noqa
+                "name": "Chicago IPCB Office",
+            }
+        elif "springfield" in text or "llinois pollution control board" in text:
+            return {
+                "address": "1021 N. Grand Ave. E. - Room 1244 N, Springfield, IL 62702",
+                "name": "Springfield IPCB Office",
+            }
+        elif "sangamo room" in text:
+            return {
+                "address": "1021 N. Grand Ave. E. - Sangamo Room, Springfield, IL 62702",
+                "name": "Illinois EPA",
+            }
+        else:
+            return {
+                "address": "",
+                "name": "",
+            }
+
+    def _parse_links(self, meeting):
+        """ Associate Meeting objects with previously-scraped links """
+        links = list()
+        key = meeting['start'].date()
+        if key in self.minutes_map:
+            links.append({"href": self.minutes_map[key], "title": "Minutes"})
+        if key in self.agenda_map:
+            links.append({"href": self.agenda_map[key], "title": "Agenda"})
+
+        return links
+
+    def _parse_source(self, item):
+        """Parse or generate source."""
+        rel_url = scrapy.Selector(text=item["Description"]).xpath(".//a/@href").get()
+        if rel_url:
+            return urljoin("https://{}".format(self.allowed_domains[0]), rel_url)
```

Comment:
Here's another spot where `response.urljoin` can be used instead as long as `response` is passed to `_parse_source`

Reply:
Alrighty! Just for my own benefit, what makes response.urljoin() preferred?

Reply:
It's mostly because we're moving away from including `allowed_domains` since it's generally unnecessary for us, but it's also more explicit about including the protocol and domain from the current response rather than hard-coding values to start